app/src/androidTest/java/ch/epfl/sdp/blindly/weather/WeatherActivityTest.kt

```diff
@@ -86,6 +99,35 @@ class WeatherActivityTest {
         Intents.release()
     }
 
+    @Test
+    fun setDayAndAddEvent() {
+        val date = Calendar.getInstance()
+        date.set(YEAR, MONTH, DAY_OF_MONTH)
+        val calIntent = Intent(Intent.ACTION_INSERT)
+        calIntent.data = CalendarContract.Events.CONTENT_URI
+        calIntent.putExtra(CalendarContract.Events.TITLE, CALENDAR_EVENT_TITLE)
+        calIntent.putExtra(CalendarContract.EXTRA_EVENT_ALL_DAY, true)
+        calIntent.putExtra(
+            CalendarContract.EXTRA_EVENT_BEGIN_TIME,
+            date.timeInMillis
+        )
+        calIntent.putExtra(
+            CalendarContract.EXTRA_EVENT_END_TIME,
+            date.timeInMillis
+        )
+
+        onView(
+            allOf(
+                withId(R.id.eventButton),
+                withParent(withId(R.id.weather_layout))
+            )
+        ).perform(click())
+        intended(IntentMatchers.hasExtraWithKey(CalendarContract.Events.TITLE))
+        intended(IntentMatchers.hasExtraWithKey(CalendarContract.EXTRA_EVENT_ALL_DAY))
+        intended(IntentMatchers.hasExtraWithKey(CalendarContract.EXTRA_EVENT_BEGIN_TIME))
+        intended(IntentMatchers.hasExtraWithKey(CalendarContract.EXTRA_EVENT_END_TIME))
```

Comment:
I think it could be interesting to import hasExtraWithKey to remove the IntentMatchers